integration/aggregate_documents_test.go

```diff
@@ -659,3 +662,248 @@ func TestAggregateUnsetErrors(t *testing.T) {
+			cursor: bson.D{{"batchSize", -1.1}},
+			err: &mongo.CommandError{
+				Code:    51024,
+				Name:    "Location51024",
+				Message: "BSON field 'batchSize' value must be >= 0, actual value '-1'",
+			},
+		},
+		"DoubleFloor": {
+			cursor:     bson.D{{"batchSize", 1.9}},
+			firstBatch: docs[:1],
+		},
+		"Bool": {
+			cursor:     bson.D{{"batchSize", true}},
+			firstBatch: docs[:1],
+			err: &mongo.CommandError{
+				Code:    14,
+				Name:    "TypeMismatch",
+				Message: "BSON field 'cursor.batchSize' is the wrong type 'bool', expected types '[long, int, decimal, double']",
+			},
+			altMessage: "BSON field 'aggregate.batchSize' is the wrong type 'bool', expected types '[long, int, decimal, double]'",
+		},
+		"Unset": {
+			cursor:     nil,
+			firstBatch: docs[:101],
+			err: &mongo.CommandError{
+				Code:    9,
+				Name:    "FailedToParse",
+				Message: "The 'cursor' option is required, except for aggregate with the explain argument",
+			},
+		},
+		"Empty": {
+			cursor:     bson.D{},
+			firstBatch: docs[:101],
+		},
+		"String": {
+			cursor:     "invalid",
+			firstBatch: docs[:101],
+			err: &mongo.CommandError{
+				Code:    14,
+				Name:    "TypeMismatch",
+				Message: "cursor field must be missing or an object",
+			},
+			altMessage: "BSON field 'cursor' is the wrong type 'string', expected type 'object'",
+		},
+		"LargeBatchSize": {
+			cursor:     bson.D{{"batchSize", 102}},
+			firstBatch: docs[:102],
+		},
+		"LargeBatchSizeMatch": {
+			pipeline: bson.A{
+				bson.D{{"$match", bson.D{{"_id", bson.D{{"$in", bson.A{0, 1, 2, 3, 4, 5}}}}}}},
+			},
+			cursor:     bson.D{{"batchSize", 102}},
+			firstBatch: docs[:6],
+		},
+	} {
+		name, tc := name, tc
+		t.Run(name, func(t *testing.T) {
+			if tc.skip != "" {
+				t.Skip(tc.skip)
+			}
+
+			t.Parallel()
+
+			var pipeline any = bson.A{}
+			if tc.pipeline != nil {
+				pipeline = tc.pipeline
+			}
+
+			var rest bson.D
+			if tc.cursor != nil {
+				rest = append(rest, bson.E{Key: "cursor", Value: tc.cursor})
+			}
+
+			command := append(
+				bson.D{
+					{"aggregate", collection.Name()},
+					{"pipeline", pipeline},
+				},
+				rest...,
+			)
+
+			var res bson.D
+			err := collection.Database().RunCommand(ctx, command).Decode(&res)
+			if tc.err != nil {
+				assert.Nil(t, res)
+				AssertEqualAltCommandError(t, *tc.err, tc.altMessage, err)
+
+				return
+			}
+
+			require.NoError(t, err)
+
+			v, ok := res.Map()["cursor"]
+			require.True(t, ok)
+
+			cursor, ok := v.(bson.D)
+			require.True(t, ok)
+
+			// do not check the value of cursor id, FerretDB has a different id
+			cursorID := cursor.Map()["id"]
+			assert.NotNil(t, cursorID)
+
+			firstBatch, ok := cursor.Map()["firstBatch"]
+			require.True(t, ok)
+			require.Equal(t, tc.firstBatch, firstBatch)
+		})
+	}
+}
+
+func TestAggregateBatchSize(t *testing.T) {
+	t.Parallel()
+	ctx, collection := setup.Setup(t)
+
+	// the number of documents is set to much bigger than the default batchSize of 101
```

Comment:
Strictly speaking, if we want to answer the question "why", in this comment, we should also mention that we want it to cause `getMore` needed to be called (if I understood the idea correctly).

Reply:
Good point, nowhere currently mention about implicit call to `getMore`. Updated the test comments